scripts/gulpfiles/build_tasks.js

```diff
@@ -672,6 +674,56 @@ function buildCompiled() {
       .pipe(gulp.dest(RELEASE_DIR));
 }
 
+/**
+ * This task builds the shims used by the playgrounds and tests to
+ * load Blockly in either compressed or uncompressed mode, creating
+ * build/blockly.mjs, blocks.mjs, javascript.mjs, etc.
+ *
+ * Prerequisite: getChunkOptions (via buildCompiled, for chunks[].parent).
+ */
+async function buildShims() {
+  // Install a package.json file in BUILD_DIR to tell node.js that the
+  // .js files therein are ESM not CJS, so we can import the
+  // entrypoints to enumerate their exported names.
+  //
+  // N.B.: There is an exception: core/main.js is a goog.module not
+  // ESM, but fortunately we don't attempt to import or require this
+  // file from node.js - we only feed it to Closure Compiler, which
+  // uses the type information in deps.js rather than package.json.
+  await fsPromises.writeFile(
+    path.join(BUILD_DIR, 'package.json'),
+    '{"type": "module"}'
+  );
+
+  // Import each entrypoint module, enumerate its exports, and write
+  // a shim to load the chunk either by importing the entrypoint
+  // module or by loading the compiled script.
+  await Promise.all(chunks.map(async (chunk) => {
+    const modulePath = posixPath(chunk.moduleEntry ?? chunk.entry);
+    const scriptPath =
+        path.posix.join(RELEASE_DIR, `${chunk.name}${COMPILED_SUFFIX}.js`);
+    const shimPath = path.join(BUILD_DIR, `${chunk.name}.mjs`);
```

Comment:
How would you feel about changing this to `${chunk.name}.loader.mjs` to make it clearer within the playgrounds that we're importing a shim?

On one hand I like how the imports in the playgrounds are very similar to our imports in samples. But on the other hand, I feel like the current naming hides the fact that we actually are using a different thing that's doing different work.

Reply:
That seems reasonable enough.